# Individual Planning Report

In [1]:
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


## Data Description

In [4]:
players <- read_csv("players.csv")
sessions <- read_csv("sessions.csv")

glimpse(players)
glimpse(sessions)

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Rows: 196
Columns: 7
$ experience   <chr> "Pro", "Veteran", "Veteran", "Amateur", "Regular", "Amate…
$ subscribe    <lgl> TRUE, TRUE, FALSE, TRUE, TRUE, TRUE, TRUE, FALSE, TRUE, T…
$ hashedEmail  <chr> "f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8…
$ played_hours <dbl> 30.3, 3.8, 0.0, 0.7, 0.1, 0.0, 0.0, 0.0, 0.1, 0.0, 1.6, 0…
$ name         <chr> "Morgan", "Christian", "Blake", "Flora", "Kylie", "Adrian…
$ gender       <chr> "Male", "Male", "Male", "Female", "Male", "Female", "Fema…
$ Age          <dbl> 9, 17, 17, 21, 21, 17, 19, 21, 47, 22, 23, 17, 25, 22, 17…
Rows: 1,535
Columns: 5
$ hashedEmail         <chr> "bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8a…
$ start_time          <chr> "30/06/2024 18:12", "17/06/2024 23:33", "25/07/202…
$ end_time            <chr> "30/06/2024 18:24", "17/06/2024 23:46", "25/07/202…
$ original_start_time <dbl> 1.71977e+12, 1.71867e+12, 1.72193e+12, 1.72188e+12…
$ original_end_time   <dbl> 1.71977e+12, 1.71867e+12, 1.72193e+12, 1.72188e+

- Players:
    - Number of Observations: 196
    - Number of Variables: 7
    - | Name of Variable | Type of Variable | Possible Issues |
  |---|---|---|
  | Age | Double | Type may need to be changed to type Integer to optimize for memory |
  | Gender | Character |
  | Name | Character |
  | Played Hours | Double | There seem to be outliers which may skew results hence data may need to be standardized |
  | Hashed Email | Character |
  | Subscribe | Logical |
  | Experience | Character |

- Sessions
    - Number of Observations: 1535
    - Number of Variables: 5
    - | Name of Variable | Type of Variable | Possible Issues |
  |---|---|---|
  | Start Time | Character | Type may need to be changed to data/time format to utilize effectively |
  | End Time | Character | Type may need to be changed to data/time format to utilize effectively |
  | Original Start Time | Double | Data may need to be scaled/changed to a more readable form |
  | Original End Time | Double | Data may need to be scaled/changed to a more readable form |
  | Hashed Email | Character |



## Questions

**Broad Question:** 
- Question 1: What player characteristics and behaviours are most predictive of subscribing to a game-related newsletter, and how do these features differ between various player types?
  
**Specific Question:** 
- Can age, gender and played hours predict whether a player subscribes to a game-related newsletter?

**Describe clearly how the data will help you address the question of interest. You may need to describe how you plan to wrangle your data to get it into a form where you can apply one of the predictive methods from this class.**
- The players.csv dataset will be used in order to address this question of interest. It contains the explanatory variables we are focusing on, which are age, gender, and hours played as well as our response variable which is whether a player subscribes to a newsletter or not. Since, the broad questions mentioned both characteristics and behaviors, we can choose these three explanatory variables to answer both sides of the question, and give an accurate representation of whether a player subscribes to the newsletter or not.

## Exploratory Data Analysis and Visualization